### DISCUSSION TITLE
What would the preferred Markdown Tag be?

### DISCUSSION THREAD

**Discussion by MarketingPip** [MarketingPipeline/Markdown-Tag#2] [General]
What would the preferred markdown tag be? Another user has mentioned it might look better as

`<mark-down flavor="common">`

as opposed to using 

`<md> and <github-md>`

seeing if there is any other possible suggestions on this & in future a poll will be held regarding if it should be changed or not. 
From there - a final decision on the tag name will be made with help (**not based alone**) from the poll data. 

Any suggestion regarding this before a poll takes place would be more than appreciated!

**Comment by shaunbharat** [comment]
Perhaps "markdown" could be used without the hyphen?

`<markdown flavor="common"`

or shorter

`<md flavor="common">`

**Comment by MarketingPip** [comment]
I do believe there is a fork being started that follows this standard, I am waiting for him to join this repo & get some further input / discussion. But I do like the shortened tag with the flavor attribute.

**Comment by DarkenLM** [comment]
As the porter of MarkdownTag to a WebComponent, any tag will have to include an hyphen in the name, as per the [spec](https://html.spec.whatwg.org/multipage/custom-elements.html#valid-custom-element-name), to be able to be used on the WebComponent. Currently, I'm using `md-tag`, but I am open to suggestions.

**Reply by MarketingPip** [reply]
Is there any way we can use  `<md flavor="common"> </md>`

**Reply by DarkenLM** [reply]
Unfortunately, no. The name has to include an hyphen.

**Reply by shaunbharat** [reply]
I think `<md-tag flavor="common"><md-tag/>` is okay, it's basically the repository name

**Reply by MarketingPip** [reply]
> Unfortunately, no. The name has to include an hyphen.

Pretty sure I have a way to do this without a **hyphen.**  I added you to another repo to discuss something off-topic from this project + I will share code on how to possibly do this.

**Reply by DarkenLM** [reply]
When using WebComponents, it is not, unfortunately. Attempting to define a Custom Element without a hyphen on it's name will throw the following error:
```
Uncaught DOMException: Failed to execute 'define' on 'CustomElementRegistry': "md" is not a valid custom element name
```

WebComponents are better to use rather than attempting to parse raw HTML, as it allows for better encapsulation and allows for us to handle work to the DOM to parse the tag, allowing for a better integration.
